src/txn/txn.c

```diff
@@ -138,6 +138,42 @@ __wt_txn_release_snapshot(WT_SESSION_IMPL *session)
     }
 }
 
+/*
+ * __wt_txn_user_active --
+ *     Check whether there are any running user transactions.
+ */
+bool
+__wt_txn_user_active(WT_SESSION_IMPL *session)
+{
+    WT_CONNECTION_IMPL *conn;
+    WT_TXN_GLOBAL *txn_global;
+    WT_TXN_SHARED *s;
+    uint32_t i, session_cnt;
+    bool txn_active;
+
+    conn = S2C(session);
+    txn_global = &conn->txn_global;
+    txn_active = true;
+
+    /* We're going to scan the table in exclusive mode: wait for the write lock. */
+    __wt_writelock(session, &txn_global->rwlock);
+
+    /* Walk the array of concurrent transactions. */
+    WT_ORDERED_READ(session_cnt, conn->session_cnt);
+    WT_STAT_CONN_INCR(session, txn_walk_sessions);
+    for (i = 0, s = txn_global->txn_shared_list; i < session_cnt; i++, s++) {
+        WT_STAT_CONN_INCR(session, txn_sessions_walked);
+        /* If session has a valid transaction id, there is at least one active transaction. */
+        if (s->id != WT_TXN_NONE && !F_ISSET(session, WT_SESSION_INTERNAL))
```

Comment:
I am not sure that this covers all the user transactions. It is possible that a transaction with no write operations cannot have the `id` to be set. Am I missing anything?

Reply:
Why we need to implement another function? Can we pass in a flag to __wt_txn_activity_check to ignore internal transactions?

Reply:
@quchenhao I looked into using a flag however there are at least 3 layers of function call that will need to know that flag. `__wt_txn_activity_check` and the helper functions are designed to assess transaction activity based on oldest id being different than the current id.

Reply:
@kommiharibabu are you talking about scenario where we have an open cursor and session outside the context of a running transaction?

I just checked that it possible to not have a transaction id yet if the transaction has not made any modifications.